test/unit/proxy_test.rb

```diff
@@ -148,6 +148,16 @@ def test_endpoint_validation
 
       service.deployment_option = 'plugin_ruby'
       assert_valid proxy
+
+      service.deployment_option = 'self_managed'
```

Comment:
By the way we wanted to move the deployment option to proxy. Another cleanup we need to do later